include/executors/reduction.hpp

```diff
@@ -0,0 +1,132 @@
+#ifndef REDUCTION_HPP
+#define REDUCTION_HPP
```

Comment:
renamed to reduction_sycl.hpp with header guard BLAS_REDUCTION_SYCL_HPP, to match other files.